pkg/controller/elasticsearch/nodespec/defaults.go

```diff
@@ -38,10 +38,15 @@ var (
 	// DefaultResources for the Elasticsearch container. The JVM default heap size is 1Gi, so we
 	// request at least 2Gi for the container to make sure ES can work properly.
 	// Not applying this minimum default would make ES randomly crash (OOM) on small machines.
+	// Similarly, we apply a default memory limit of 2Gi, to ensure the Pod isn't the first one to get evicted.
+	// No CPU requirement is set by default.
 	DefaultResources = corev1.ResourceRequirements{
 		Requests: map[corev1.ResourceName]resource.Quantity{
 			corev1.ResourceMemory: resource.MustParse("2Gi"),
 		},
+		Limits: map[corev1.ResourceName]resource.Quantity{
+			corev1.ResourceMemory: resource.MustParse("2Gi"),
+		},
```

Comment:
Technically, we don't have to apply both Requests and Limits: if only Limits are set, Kubernetes defaults to set Requests to the same value.
However I tend to prefer making both clearly explicit here (the comment helps understanding their respective role).